xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/VideoFilterShaderGLES.cpp

```diff
@@ -0,0 +1,302 @@
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "system.h"
+
+#include <string>
+#include <math.h>
+
+#include "VideoFilterShaderGLES.h"
+#include "utils/log.h"
+#include "utils/GLUtils.h"
+#include "ConvolutionKernels.h"
+#include "windowing/WindowingFactory.h"
+
+#define TEXTARGET GL_TEXTURE_2D
+
+using namespace Shaders;
+
+//////////////////////////////////////////////////////////////////////
+// BaseVideoFilterShader - base class for video filter shaders
+//////////////////////////////////////////////////////////////////////
+
+BaseVideoFilterShader::BaseVideoFilterShader()
+{
+  m_width = 1;
+  m_height = 1;
+  m_hStepXY = 0;
+  m_stepX = 0;
+  m_stepY = 0;
+  m_sourceTexUnit = 0;
+  m_hSourceTex = 0;
+
+  m_stretch = 0.0f;
+
+  m_hVertex = -1;
+  m_hcoord = -1;
+  m_hProj = -1;
+  m_hModel = -1;
+  m_hAlpha = -1;
+
+  m_proj = nullptr;
+  m_model = nullptr;
+  m_alpha = -1;
+
+  std::string shaderv =
+    "attribute vec4 m_attrpos;"
+    "attribute vec2 m_attrcord;"
+    "varying vec2 cord;"
+    "uniform mat4 m_proj;"
+    "uniform mat4 m_model;"
+
+    "void main ()"
+    "{"
+    "mat4 mvp = m_proj * m_model;"
+    "gl_Position = mvp * m_attrpos;"
+    "cord = m_attrcord.xy;"
+    "}";
+  VertexShader()->SetSource(shaderv);
+
+  std::string shaderp =
+    "precision mediump float;"
+    "uniform sampler2D img;"
+    "varying vec2 cord;"
+    "void main()"
+    "{"
+    "gl_FragColor = texture2D(img, cord);"
+    "}";
+  PixelShader()->SetSource(shaderp);
+}
+
+void BaseVideoFilterShader::OnCompiledAndLinked()
+{
+  m_hVertex = glGetAttribLocation(ProgramHandle(),  "m_attrpos");
+  m_hcoord = glGetAttribLocation(ProgramHandle(),  "m_attrcord");
+  m_hAlpha  = glGetUniformLocation(ProgramHandle(), "m_alpha");
+  m_hProj  = glGetUniformLocation(ProgramHandle(), "m_proj");
+  m_hModel = glGetUniformLocation(ProgramHandle(), "m_model");
+}
+
+bool BaseVideoFilterShader::OnEnabled()
+{
+  glUniformMatrix4fv(m_hProj,  1, GL_FALSE, m_proj);
+  glUniformMatrix4fv(m_hModel, 1, GL_FALSE, m_model);
+  glUniform1f(m_hAlpha, m_alpha);
+  return true;
+}
+
+ConvolutionFilterShader::ConvolutionFilterShader(ESCALINGMETHOD method, bool stretch, GLSLOutput *output)
+{
+  m_method = method;
+  m_kernelTex1 = 0;
+  m_hKernTex = -1;
+
+  std::string shadername;
+  std::string defines;
+
+  if (g_Windowing.IsExtSupported("GL_EXT_color_buffer_float"))
+  {
+    m_floattex = true;
+  }
+  else
+  {
+    m_floattex = false;
+  }
+
+  if (m_method == VS_SCALINGMETHOD_CUBIC ||
+      m_method == VS_SCALINGMETHOD_LANCZOS2 ||
+      m_method == VS_SCALINGMETHOD_SPLINE36_FAST ||
+      m_method == VS_SCALINGMETHOD_LANCZOS3_FAST)
+  {
+    shadername = "convolution-4x4.glsl";
+  }
+  else if (m_method == VS_SCALINGMETHOD_SPLINE36 ||
+           m_method == VS_SCALINGMETHOD_LANCZOS3)
+  {
+    shadername = "convolution-6x6.glsl";
+  }
+
+  if (m_floattex)
+  {
+    m_internalformat = GL_RGBA16F_EXT;
```

Comment:
fails on RBPI beacause `GL_RGBA16F_EXT` isn't declared

Reply:
RBPI builds should not include this file

Reply:
why not?

Reply:
it only uses its own renderer